moped-editor/src/components/FallbackComponent.js

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { Box, Container, Typography } from "@mui/material";
+import makeStyles from "@mui/styles/makeStyles";
+import ExternalLink from "./ExternalLink";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    height: "100%",
+    paddingBottom: theme.spacing(3),
+    paddingTop: theme.spacing(3),
+    backgroundColor: theme.palette.background.paper,
+    maxWidth: "525px",
+  },
+  image: {
+    display: "inline-block",
+    maxWidth: "100%",
+    width: 200,
+  },
+  subheading: {
+    marginTop: "1rem",
+  },
+  errorText: {
+    backgroundColor: theme.palette.background.summaryHover,
+    fontSize: ".75rem",
+    padding: "8px",
+    fontFamily: "monospace",
+  },
+}));
+
+const createBugReportLink = (error) =>
+  `https://atd.knack.com/dts#new-service-request/?view_249_vars=%7B%22field_398%22%3A%22Bug%20Report%20%E2%80%94%20Something%20is%20not%20working%22%2C%22field_399%22%3A%22Moped%22%2C%22field_400%22%3A%22${error.message}%22%7D`;
+
+const FallbackComponent = ({ error, resetErrorBoundary }) => {
+  const classes = useStyles();
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      height="100%"
```

Comment:
I read y'all's latest thoughts on this, and I am also split on how high or low in the tree to go with the error boundaries. It is nice to keep the rest of the app going when an error occurs somewhere deeper in the main layout, but that creates decisions to be made about where to place them. 🤔 Maybe the top level of the main views/routes at least? Besides those, the components map jumps out at me since there are a lot of moving pieces, and it is hard to test.

I also noticed that this library has a `resetErrorBoundary` function to reset and try again with a button. I think that would have the same effect as refreshing the page (like the error component says to do) in a lot of our cases, though.

Last thought, we might consider the type of errors that we aim to catch and how/if that could intersect with other ways of catching code errors like TS.